src/main/java/io/stargate/sgv2/jsonapi/api/v1/CollectionResource.java

```diff
@@ -164,12 +168,23 @@ else if (error instanceof JsonApiException jsonApiException) {
                 // otherwise use generic for now
                 return Uni.createFrom().item(new ThrowableCommandResultSupplier(error));
               } else {
+                EmbeddingService embeddingService = null;
+                if (collectionProperty.vectorizeServiceName() != null
+                    && collectionProperty.modelName() != null) {
+                  embeddingService =
```

Comment:
So we always fetch embeddingService if the Collection has one configured? Although we don't yet know if one is needed?

Reply:
Yes, Embedding service is light weight object which is cached.